common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java

```diff
@@ -54,6 +54,98 @@
      */
     private String mFamilyId;
 
+    /**
+     * The deployment ring of the current request chain.
+     */
+    private String mSpeRing;
+
+    /**
+     * The age of the RT, according to the server.
+     */
+    private String mRefreshTokenAge;
+
+    /**
+     * The error code surfaced by the server.
+     */
+    private String mServerErrorCode;
+
+    /**
+     * The server suberror code.
+     */
+    private String mServerSubErrorCode;
+
+    /**
+     * Gets the SPE Ring.
+     *
+     * @return The SPE Ring to get.
+     */
+    public String getSpeRing() {
+        return mSpeRing;
+    }
+
+    /**
+     * Sets the SPE Ring.
+     *
+     * @param speRing The SPR Ring to set.
+     */
+    public void setSpeRing(String speRing) {
+        this.mSpeRing = speRing;
+    }
+
+    /**
+     * Gets the refresh token age.
+     *
+     * @return The refresh token age to get.
+     */
+    public String getRefreshTokenAge() {
+        return mRefreshTokenAge;
+    }
+
+    /**
+     * Sets the refresh token age.
+     *
+     * @param refreshTokenAge The refresh token age to set.
+     */
+    public void setRefreshTokenAge(String refreshTokenAge) {
+        this.mRefreshTokenAge = refreshTokenAge;
+    }
+
+    /**
+     * Gets the server error code.
+     *
+     * @return The server error code to get.
+     */
+    public String getServerErrorCode() {
```

Comment:
Q: should these be in MicrosoftTokenErrorResponse?

Reply:
Should these also be in part of MicrosoftTokenErrorResponse?

Reply:
Synced offline - the result is moved by the `LocalAuthenticationResult` or via `BaseException` (if things go wrong)